tools/dependency/release_dates.py

```diff
@@ -48,22 +56,52 @@ def VerifyAndPrintLatestRelease(dep, repo, metadata_version, release_date):
 def VerifyAndPrintReleaseDate(dep, github_release_date, metadata_release_date):
   mismatch = ''
   iso_release_date = FormatUtcDate(github_release_date)
-  print(f'{dep} has a GitHub release date {iso_release_date}')
+  Colorize("GREEN", f'{dep} has a GitHub release date {iso_release_date}')
   if iso_release_date != metadata_release_date:
     raise ReleaseDateError(f'Mismatch with metadata release date of {metadata_release_date}')
 
 
 # Extract release date from GitHub API.
-def GetReleaseDate(repo, metadata_version, github_release):
+def GetReleaseDate(dep, repo, metadata_version, github_release):
   if github_release.tagged:
-    tags = repo.get_tags()
-    for tag in tags:
-      if tag.name == github_release.version:
-        return tag.commit.commit.committer.date
+
+    # Not all Github repositories with releases have a latest release
+    try:
+      latest = repo.get_latest_release()
+    except github.UnknownObjectException:
+      latest = None
+
+    # When the repository does have a latest release, compare the metadata version with the latest
+    # tag. In cases where a repository has a release that is later than the latest release (eg.
```

Comment:
Did you mean "has a tag that is later than the latest release"?

Reply:
Done